Library/Homebrew/extend/git_repository.rb

```diff
@@ -40,11 +40,12 @@ def git_head
   end
 
   # Gets a short commit hash of the HEAD commit.
-  sig { returns(T.nilable(String)) }
-  def git_short_head
+  sig { params(length: T.nilable(Integer)).returns(T.nilable(String)) }
+  def git_short_head(length: 4)
     return unless git? && Utils::Git.available?
```

Comment:
I know you didn't write this but: if this could be refactored to not be `unless .. &&` and instead an `if` I'd be grateful 🙇🏻. For bonus points in another PR: I'd love a `rubocop` that checks for `unless .. &&` and `unless ... ||`. https://github.com/rubocop-hq/rubocop/pull/5400 was a first attempt for this in RuboCop itself but I think it'd be fine for this to just be a local cop for Homebrew until/if you're able to upstream it.